### ISSUE TITLE
the loss values are negative and fluctuating without any clear pattern

### ISSUE THREAD

**Issue by JasonYpro** [zhaodongsun/contrast-phys#1]
Hello! I would like to inquire about replicating your paper using the UBFC dataset. I have set the number of epochs to 30, while keeping the other parameters the same as those provided in your paper. However, I noticed that the loss values are negative and fluctuating without any clear pattern. Is this behavior normal?
![image](https://github.com/zhaodongsun/contrast-phys/assets/79305522/97ef5a43-f08e-4d2c-9378-59ccf13a5fe7)

**Comment by zhaodongsun** [comment]
Hi, The loss value can be negative. You can smooth the loss curve by exponential smoothing to see the decreasing trend.

**Comment by JasonYpro** [comment]
Hello, I still can't solve the above problem (loss function curve jittering too much), and I'm not sure if this can be interpreted as an overfitting problem. In the meantime, I trained a model through the UBFC dataset to compare with the .pt file provided according to your demo, using subject38 for testing. The first graph is my test result and the second graph is the weights provided by you, you can see that the curves are similar, but the value of the vertical axis of my result is 10 times smaller compared to your model, put the two results together as shown in figure 3. All parameters are consistent with the code you provided, maybe my training process is wrong, can you explain this phenomenon?
![image](https://github.com/zhaodongsun/contrast-phys/assets/79305522/adf1d837-4e95-4701-a024-290b1569f4f2)
![image](https://github.com/zhaodongsun/contrast-phys/assets/79305522/2bbae483-6bde-4e76-b2cc-a350873fe4e6)
![image](https://github.com/zhaodongsun/contrast-phys/assets/79305522/47cc5f60-16ca-40c1-9c5e-ab2fbfd8da7d)

**Comment by zhaodongsun** [comment]
Hi, I haven't noticed this phenomenon before. My assumption is that the model does not explicitly learn the signal amplitude, so the amplitude may vary during different training. However, the signal amplitude is not important since the scaled amplitude does not influence the following analysis such as HR or HRV calculation. Usually, the signal amplitude should be normalized with zero mean and one std.

**Comment by dalaoplan** [comment]
> Hello! I would like to inquire about replicating your paper using the UBFC dataset. I have set the number of epochs to 30, while keeping the other parameters the same as those provided in your paper. However, I noticed that the loss values are negative and fluctuating without any clear pattern. Is this behavior normal? ![image](https://private-user-images.githubusercontent.com/79305522/277144979-97ef5a43-f08e-4d2c-9378-59ccf13a5fe7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDM3NTM1NjEsIm5iZiI6MTcwMzc1MzI2MSwicGF0aCI6Ii83OTMwNTUyMi8yNzcxNDQ5NzktOTdlZjVhNDMtZjA4ZS00ZDJjLTkzNzgtNTljY2YxM2E1ZmU3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjI4VDA4NDc0MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTA5YTRjNzE4ZGU1OWI4Yzg0YzAyYWQxZjI3NzM3OTFhMmFmMDliMTE3MTg5ZjQyNzVlMTMwZmY5ZDczNzM3NzImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.KrdM0fMmyQDJx_5eUo9qW86w8nntmY_BJJoYVtxy6qM)



> Hello! I would like to inquire about replicating your paper using the UBFC dataset. I have set the number of epochs to 30, while keeping the other parameters the same as those provided in your paper. However, I noticed that the loss values are negative and fluctuating without any clear pattern. Is this behavior normal? ![image](https://private-user-images.githubusercontent.com/79305522/277144979-97ef5a43-f08e-4d2c-9378-59ccf13a5fe7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDM3NTQwNDMsIm5iZiI6MTcwMzc1Mzc0MywicGF0aCI6Ii83OTMwNTUyMi8yNzcxNDQ5NzktOTdlZjVhNDMtZjA4ZS00ZDJjLTkzNzgtNTljY2YxM2E1ZmU3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjI4VDA4NTU0M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWIwNWY0MTM1MDM5N2NjZDNkNTE0NTYzZTIyZjZlOWRlYjc4ZDgzYTc1ODZiNmVlZWIwNDg5N2QxNzQyOWQ0OTkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.2RM2P99qMQp4x_bqOV-rhNdoc-XotX_Yw1t7WCUfTNg)

**Comment by dalaoplan** [comment]
@Harry2345 Hello, I am also reproducing the project code.Running the train.py file results in an error saying that the .h5 file does not have a bvp file.  How can this be resolved?  Can you provide an example or a solution?I hope to exchange ideas!!!

**Comment by dalaoplan** [comment]
> 你好，我以前没有注意到这个现象。我的假设是模型没有明确学习信号幅度，因此在不同的训练过程中幅度可能会有所不同。然而，信号幅度并不重要，因为缩放幅度不会影响后续分析，例如 HR 或 HRV 计算。通常，信号幅度应使用零平均值和一个标准值进行归一化。

@zhaodongsun  I encountered a similar problem. I used `object 6,8,9` on the PURE data set to verify. The results are as shown below, but the visualization results show that rPPG is much smaller than ground true. You mentioned using zero average and normalized to a standard value. I don't understand. Is there any specific code?
**MAE:0.3529411764705878
MSE:0.35294117647058737
RMSE:0.5940885257860039
R:0.9988294253545278**
![06-01](https://github.com/zhaodongsun/contrast-phys/assets/74132798/e666a188-1026-49dd-b345-ce404aaa18f5)
![06-03](https://github.com/zhaodongsun/contrast-phys/assets/74132798/7403977c-1189-4745-bb57-0cd85932b3e7)

**Comment by dalaoplan** [comment]
> Hi, I haven't noticed this phenomenon before. My assumption is that the model does not explicitly learn the signal amplitude, so the amplitude may vary during different training. However, the signal amplitude is not important since the scaled amplitude does not influence the following analysis such as HR or HRV calculation. Usually, the signal amplitude should be normalized with zero mean and one std.

@zhaodongsun Hi, I encountered a similar problem. I used object 6,8,9 on the PURE data set to verify. The results are as shown below, but the visualization results show that rPPG is much smaller than ground true. You mentioned using zero average and normalized to a standard value. I don't understand. Is there any specific code?
**MAE:0.3529411764705878
MSE:0.35294117647058737
RMSE:0.5940885257860039
R:0.9988294253545278**
![06-01](https://github.com/zhaodongsun/contrast-phys/assets/74132798/d01ab108-c5a4-463f-acf5-95dce9267acb)
![06-03](https://github.com/zhaodongsun/contrast-phys/assets/74132798/674a385b-87dc-444e-99e2-d8862b143bfd)

**Comment by zhaodongsun** [comment]
@dalaoplan Hi, you can check the `normalize` function in `utils_sig.py`. Normalization is only for better visualization, but does not influence the HR results. If you want to plot two signals and compare them, you should normalize them for better visualization.

**Comment by dalaoplan** [comment]
> @dalaoplan Hi, you can check the `normalize` function in `utils_sig.py`. Normalization is only for better visualization, but does not influence the HR results. If you want to plot two signals and compare them, you should normalize them for better visualization.

@zhaodongsun Thank you, I missed it and it turned out to be in the code!
